src/common/transformations/src/transformations/mlir/convert.cpp

```diff
@@ -294,7 +299,23 @@ void injectMLIR(std::shared_ptr<ov::Model> model, MLIRContext* context, bool tpp
     model->validate_nodes_and_infer_types();
 }
 
+void loadDialects(MLIRContext* context) {
+    context->loadDialect<mlir::func::FuncDialect>();
+    context->loadDialect<mlir::linalg::LinalgDialect>();
+    context->loadDialect<mlir::bufferization::BufferizationDialect>();
+}
 
+#ifdef GRAPH_COMPILER
+MLIRContext* get_shared_mlir_context(bool ignore) {
+    static MLIRContext* context = []() {
+        OPENVINO_MLIR_DEBUG_PRINT("[ DEBUG ] Using GraphCompiler\n");
+        static MLIRContext context(gc::initCompilerAndGetDialects());
+        loadDialects(&context);
+        return &context;
+    }();
+    return context;
+}
+#else
```

Comment:
Could you activate the graph compiler path in run-time similarly how it is done for tpp-mlir? Via `OV_MLIR_GRAPH_COMPILER`. Just to have ability to compare different modes in run-time without re-building.

Reply:
May I introduce a new env var - OV_MLIR that could be either TPP or GC? In this case we avoid the error.

Reply:
Correct.